CONTRIBUTING.md

```diff
@@ -0,0 +1,56 @@
+# OpenTelemetry Contributor Guide
+
+Welcome to OpenTelemetry! This document is the single source of truth for how to
```

Comment:
```suggestion
Welcome to OpenTelemetry! This document provides general guidance for contributing to 
this codebase. This is a multi-lingual codebase, so sub-directories like `collector` may
provide additional guidance in a CONTRIBUTING.md.
```

The CONTRIBUTING.md in `opentelemetry-collector-contrib` has a lot more prescriptive detail which I think is quite valuable. The trouble is that they get into dev workflows and we probably shouldn't do that in the top-level document.

My thought in making this change in the language is that we could start by adding a CONTRIBUTING.md to the `collector` when time allows and follow up with additional consultation. I've noticed with PRs in the Collector that guidelines in that CONTRIBUTING.md could improve reviews and code quality.